README.md

```diff
@@ -1,2 +1,66 @@
+
+[WebAssembly Systems Interface (Wasi)](https://wasi.dev/) is part of the movement to use WebAssembly in the server and our goal is to put this into practice and bring it to the Kubernetes environment and enrich the krustlet ecosystem, taking advantage from the “compiled once, and run anywhere” lemma.
+
+## WebAssembly Limitations
+
+### Async functions
+
+For now, Wasm and Wasi don't have a support for asynchronous and multithreading functions, so the discovery handler had to adapted to be strictly synchronously.
+
+### Network components
+
+As mentioned before, Wasm can't handle Async calls, making it challenging to deal with networks. The current state of wasi development doesn’t include native support for Sockets and Http requests, but the community have created some workarounds that enabled some of these features. These are experimental and not recommended to be used in a production environment but serve as a preview and experience on what using the network on WebAssembly might look like before it is implemented and integrated into the main project.
+
+## Wasi Debug Echo
+To create a WebAssembly version of the Debug Echo Discovery Handler and assuming the limitations listed this version of akri has been designed to be executed in a single-threaded environment and only uses files as the communication platform, avoiding dealing with network components.
+
+### How to build and run
+
+This is a WebAssembly project, so its compilation should be targeting the wasm32-wasi architecture.
+
+```
+cargo build --target wasm32-wasi --release
+```
+
+You can use [wasmtime](https://wasmtime.dev/) to run it locally, since wasm modules are sandboxed, we need to make sure to specify the right storage so it can have access to it during runtime, this project uses `tmp/wde-dir` as the storge directory.
+
+```
+wasmtime target/wasm32-wasi/release/wasi-debug-echo.wasm --dir ~/../../tmp/wde-dir 
+```
+
+The [wasm-to-oci](https://github.com/engineerd/wasm-to-oci) project needs to be used in order To deploy this wasm application to a container registry of your choice. This can later be used to pull the project image into a Kubernetes deployment.
+> Keep in mind that Docker Hub currently doesn’t support these types of images to their registry as it uses an unknown format, so another provider is recommended.
+
+```
+wasm-to-oci push target/wasm32-wasi/release/wasi-debug-echo.wasm  {YOUR_OCI_REGISTRY}.
+```
+
+## Discovery Handler gRPC proxy
+Since the network components from the Discovery Handler were abstracted this proxy created to execute as a container intermediate the communication between the Akri Agent and the Wasi Debug Echo. This proxy sends to an input file the discovery details sent by the agent so the wasm discovery handler can start discovering the devices, these devices are then received by this proxy in an output file, which is then sent back to the Akri Agent.
```

Comment:
If we have a two node cluster,  where one node uses regular kubelet and the second node is using krustlet, where is the Akri agent running that the proxy communicates with? Are there two agents on the first node, one for the first node and second for the krustlet node?

Reply:
From the diagram below, it would seem there is only a single akri agent. So does it mean that the krustlet based discovery handler is reporting status as the kubelet node wrt akri?